### PR TITLE
 Bug 1988905: External mode deployments fails on parsing json in ODF wizard

### PR DESCRIPTION
**Pull request by afreen23**
- fixed the failure in parsing json
 - updated review page to show only details relevant to external RHCS mode
 - updated error handling since the errors were uncaught
 - disabled rhcs mode on non supported infra

Signed-off-by: Afreen Rahman <afrahman@redhat.com>

### CODE REVIEW

File: frontend/packages/ceph-storage-plugin/src/components/create-storage-system/create-storage-system-steps/backing-storage-step/backing-storage-step.tsx [hunk 1]
```diff
@@ -113,18 +115,21 @@ export const BackingStorage: React.FC<BackingStorageProps> = ({
...
-  );
+  const infraType = infra?.spec?.platformSpec?.type;
+  const enableRhcs = RHCS_SUPPORTED_INFRA.includes(infraType);
+
+  const allowedExternalStorage: ExternalStorage[] =
+    !enableRhcs || hasOCS
+      ? SUPPORTED_EXTERNAL_STORAGE.filter(
+          ({ model }) => getStorageSystemKind(model) !== STORAGE_CLUSTER_SYSTEM_KIND,
+        )
+      : SUPPORTED_EXTERNAL_STORAGE;
```
Comment (afreen23) [inline frontend/packages/ceph-storage-plugin/src/components/create-storage-system/create-storage-system-steps/backing-storage-step/backing-storage-step.tsx:132]: dont show RHCS is one is created and platform is invalid.
This is legacy, we missed it.

File: frontend/packages/ceph-storage-plugin/src/components/create-storage-system/create-storage-system-steps/backing-storage-step/backing-storage-step.tsx [hunk 2]
```diff
@@ -23,6 +23,8 @@ import { NO_PROVISIONER } from '../../../../constants';
 import './backing-storage-step.scss';
 import { GUARDED_FEATURES } from '../../../../features';
 
+const RHCS_SUPPORTED_INFRA = ['BareMetal', 'None', 'VSphere', 'OpenStack', 'oVirt', 'IBMCloud'];
```
Comment (SanjalKatiyar) [inline frontend/packages/ceph-storage-plugin/src/components/create-storage-system/create-storage-system-steps/backing-storage-step/backing-storage-step.tsx:26]: We can add this to constants file.

File: frontend/packages/ceph-storage-plugin/src/components/create-storage-system/create-storage-system-steps/review-and-create-step/review-and-create-step.tsx [hunk 3]
```diff
@@ -35,13 +36,17 @@ export const ReviewAndCreate: React.FC<ReviewAndCreateProps> = ({ state }) => {
...
+    createStorageClass,
     nodes,
   } = state;
   const { capacity, arbiterLocation, enableArbiter } = capacityAndNodes;
   const { encryption, kms, networkType } = securityAndNetwork;
   const { deployment, externalStorage, type } = backingStorage;
 
   const isMCG = deployment === DeploymentType.MCG;
+  const isRhcs = !_.isEmpty(connectionDetails);
+  const isStandaloneExternal = hasOCS && !_.isEmpty(createStorageClass);
```
Comment (SanjalKatiyar) [inline frontend/packages/ceph-storage-plugin/src/components/create-storage-system/create-storage-system-steps/review-and-create-step/review-and-create-step.tsx:49]: One doubt... we also have already set flags to check whether it is external cluster or not, will that not work here ??

File: frontend/packages/ceph-storage-plugin/src/components/create-storage-system/create-storage-system-steps/backing-storage-step/backing-storage-step.tsx (hunk 2 above)
Comment (afreen23) [reply frontend/packages/ceph-storage-plugin/src/components/create-storage-system/create-storage-system-steps/backing-storage-step/backing-storage-step.tsx:26]: Its just used. lets keep it.

File: frontend/packages/ceph-storage-plugin/src/components/create-storage-system/create-storage-system-steps/review-and-create-step/review-and-create-step.tsx (hunk 3 above)
Comment (afreen23) [reply frontend/packages/ceph-storage-plugin/src/components/create-storage-system/create-storage-system-steps/review-and-create-step/review-and-create-step.tsx:49]: If independent mode is true then we can say its rhcs but if its false we cannot say -
1. a non rhcs cluster created or
2. a non rhcs cluster not present

Also in future we will have many storage systems, so keeping this way, would not complicate

File: frontend/packages/ceph-storage-plugin/src/components/create-storage-system/external-storage/red-hat-ceph-storage/index.tsx [hunk 4]
```diff
@@ -1,106 +1,125 @@
...
-        isLoading={formState.isLoading}
-        dropzoneProps={{
-          accept: '.json',
-          onDropRejected: handleFileRejected,
-          onDropAccepted: handleFileAccepted,
-        }}
-        validated={formState.isRejected ? 'error' : 'default'}
-      />
-    </FormGroup>
+    <ErrorHandler error={podsLoadError || csvLoadError} loaded={podsLoaded || csvLoaded}>
```
Comment (SanjalKatiyar) [inline frontend/packages/ceph-storage-plugin/src/components/create-storage-system/external-storage/red-hat-ceph-storage/index.tsx:71]:
```suggestion
    <ErrorHandler error={podsLoadError || csvLoadError} loaded={podsLoaded && csvLoaded}>
```

File: frontend/packages/ceph-storage-plugin/src/components/create-storage-system/external-storage/red-hat-ceph-storage/index.tsx [hunk 5]
```diff
@@ -1,106 +1,125 @@
...
+                upload the results (JSON).
+              </Trans>{' '}
+              {downloadFile && (
+                <a
+                  id="downloadAnchorElem"
+                  href={downloadFile}
+                  download="ceph-external-cluster-details-exporter.py"
+                  target="_blank"
+                  rel="noopener noreferrer"
+                  className=""
```
Comment (SanjalKatiyar) [inline frontend/packages/ceph-storage-plugin/src/components/create-storage-system/external-storage/red-hat-ceph-storage/index.tsx:90]:
```suggestion
```
nits

**Pull request merged**